### DISCUSSION TITLE
Accessing Geometry

### DISCUSSION THREAD

**Discussion by hamilton-eng** [ansys/pyansys-geometry#919] [Q&A] [answered]
### I'm having issues accessing geometry.... does anyone know if this is because I am using a student license?

I downloaded Ansys this week so I have the 2024 version.

CODE 

```
from ansys.geometry.core import launch_modeler_with_spaceclaim

modoler_discovery= launch_modeler_with_spaceclaim()
```
----------------------------------------------------------------------------------------------------------------------------------
ERROR

> CRITICAL -  -  logger - handle_exception - Uncaught exception
> Traceback (most recent call last):
>   File "File_being_ran", line 24, in
>     modoler_discovery= launch_modeler_with_spaceclaim()
>                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>   File "C:\Users\Users\miniconda3\Lib\site-packages\ansys\geometry\core\connection\launcher.py", line 566, in launch_modeler_with_spaceclaim   
>     return prepare_and_start_backend(
>            ^^^^^^^^^^^^^^^^^^^^^^^^^^
>   File "C:\Users\Users\miniconda3\Lib\site-packages\ansys\geometry\core\connection\product_instance.py", line 236, in prepare_and_start_backend
>     _check_minimal_versions(product_version)
>   File "C:\Users\Users\miniconda3\Lib\site-packages\ansys\geometry\core\connection\product_instance.py", line 373, in _check_minimal_versions  
>     raise SystemError(msg)
> SystemError: PyAnsys Geometry is compatible with Ansys Products from version 23.2.1. Please install Ansys products 23.2.1 or later.

---------------------------------------------------------------------------------------------------------------------------------
`pip show ansys.geometry.core`

> Name: ansys-geometry-core
> Version: 0.4.3
> Summary: A python wrapper for Ansys Geometry service
> Home-page:
> Author:
> Author-email: "ANSYS, Inc." <pyansys.core@ansys.com>
> License:
> Location: C:\Users\User\miniconda3\Lib\site-packages
> Requires: ansys-api-geometry, ansys-tools-path, beartype, google-api-python-client, googleapis-common-protos, grpcio, grpcio-health-checking, numpy, Pint, protobuf, pyvista, scipy, six, vtk
> Required-by: pyansys
>

**Comment by RobPasMue** [comment]
Hi @hamilton-eng! Thanks for opening this discussion.

Could you run the following code snippet?

```py
from ansys.tools.path import get_available_ansys_installations, get_latest_ansys_installation
installs = get_available_ansys_installations()
print(installs)
latest = get_latest_ansys_installation()
print(latest)
```

This will allow us to verify your Ansys installations as well as the latest one.

**Reply by hamilton-eng** [reply]
{-232: 'C:\\Program Files\\ANSYS Inc\\ANSYS Student\\v232'}
(-232, 'C:\\Program Files\\ANSYS Inc\\ANSYS Student\\v232')

**Reply by RobPasMue** [reply]
Hi @hamilton-eng as you can see, this is not the 24R1 Ansys version - this Ansys version will be released in the following weeks.

Can you try with the latest ``0.3.X`` version of ``ansys-geometry-core`` in the meantime?

**Reply by Eddysca** [reply]
Good evening @RobPasMue,

I hope your week is off to a good start.

I am experiencing the same issue described above. At the moment I have installed Ansys 2023R3 student version (same as above), PyAnsys 2023.2.0 and "ansys-geomety-core" 0.2.10. I am experiencing the same issue as described above:

CRITICAL -  -  logger - handle_exception - Uncaught exception
Traceback (most recent call last):
  File "C:\Users\Blake\OneDrive\Documents\Ansys CFD\Python scripts\PyAnsys_Test.py", line 12, in <module>
    modeler_discovery = launch_modeler_with_spaceclaim()
  File "C:\Users\Blake\AppData\Local\Programs\Python\Python310\lib\site-packages\ansys\geometry\core\connection\launcher.py", line 539, in launch_modeler_with_spaceclaim
    return prepare_and_start_backend(
  File "C:\Users\Blake\AppData\Local\Programs\Python\Python310\lib\site-packages\ansys\geometry\core\connection\product_instance.py", line 202, in prepare_and_start_backend
    _check_minimal_versions(product_version)
  File "C:\Users\Blake\AppData\Local\Programs\Python\Python310\lib\site-packages\ansys\geometry\core\connection\product_instance.py", line 316, in _check_minimal_versions
    raise SystemError(msg)
SystemError: PyAnsys Geometry is compatible with Ansys Products from version 23.2.1. Please install Ansys products 23.2.1 or later.

Many thanks in advance for the help!

Best regards,

Edoardo

**Reply by hamilton-eng** [reply]
Name: ansys-geometry-core
Version: 0.3.2


Download this version instead

**Reply by Eddysca** [reply]
Apologies, I'm currently using 0.3.2

**Reply by hamilton-eng** [reply]
I have PyAnsys 2024.1.0 if that helps- maybe update PyAnsys?

**Reply by Eddysca** [reply]
I have upgraded the PyAnsys version to 2024.1.0 but the error is still the same. Do you have any other suggestions?

**Reply by Eddysca** [reply]
Did you manage to solve this issue at the end on your side? If yes how did you do it? Upgrading Ansys to 2024R1?

**Reply by RobPasMue** [reply]
Hi @Eddysca - thanks for commenting on this thread. Couple of things.

Version 0.2.X should work with Ansys 2023R2, however I recommend moving to version 0.3.X of PyAnsys Geometry as @hamilton-eng suggested.

Can you also run the previous script I shared with @hamilton-eng?

```py
from ansys.tools.path import get_available_ansys_installations, get_latest_ansys_installation
installs = get_available_ansys_installations()
print(installs)
latest = get_latest_ansys_installation()
print(latest)
```

**Reply by RobPasMue** [reply]
You might also be missing the Service Pack 1 for Ansys 2023R2. As you can see in the error message, it says:

```console
SystemError: PyAnsys Geometry is compatible with Ansys Products from version 23.2.1. Please install Ansys products 23.2.1 or later.
```

2023.2.1 stands for Ansys 2023R2 Service Pack 1 - sorry for the inconveniences!

**Reply by Eddysca** [reply]
Hi @RobPasMue, thanks again for the kind reply. If I run the script it shows up the same version as @hamilton-eng. As I said on the other thread I can confirm that I am missing the Service Pack 1 installation so thanks for pointing that out! Is there a way to get the Service Pack 1 for the Ansys student version?

At the same time I am waiting for my university to install Ansys 2023R1 on the system.

**Reply by RobPasMue** [reply]
Let's keep the conversation in this thread =) I believe that if you install the latest available Ansys Student version from the Ansys webpage, you will be getting SP1. Try it out and let me know! Otherwise, Ansys 24R1 will be released shortly, maybe you can wait a couple extra days and have access to the features on ``ansys-geometry-core`` version 0.4.X =)

**Reply by Eddysca** [reply]
Hi @RobPasMue, thanks again for the fast reply. Yesterday I have tried to install the R2 version but I think SP1 was not there, do I need to set it up separately or is there another download to be done? Thanks in advance!

**Reply by RobPasMue** [reply]
Hi @Eddysca - sorry, I missed your reply!!

Ansys 24R1 is out (also as a student version)  so you can try installing it now. I also realized that we were ignoring the Student installations (my bad... really sorry). I just submitted a PR to fix this : https://github.com/ansys/pyansys-geometry/pull/938

I will get this merged and released ASAP so that you can benefit from it.

About the SP1 - I double checked and it seems that the Student Versions are typically not updated with SP* , so it won't work with your 23R2 installation, sorry for the inconveniences

**Reply by RobPasMue** [reply, marked as answer]
``ansys-geometry-core`` version 0.4.6 is out! This one should be compatible with the Ansys Student version for 24R1. Let me know if you have any issues with it.

**Reply by RobPasMue** [reply]
Hi @Eddysca - have you given it a try again? We released version 0.4.8 recently which should also be compatible. Let me know if you have any further questions. Otherwise I will close this thread.
